### PR TITLE
added sort and filter button modals

### PR DESCRIPTION
**Pull request by meanwhile7**
## Issue:
- #541 
### PRD:
- https://docs.google.com/document/d/1El7k9XrwWIoR3I-7udS-cg0XxVb3InjOU4Cw3qK565g
### Design Doc:
- https://docs.google.com/document/d/1n0pcE9t2ecVmG3q7_u2S9T26SdUp2pfEuDDGG5i_TzM/edit
## Description
UI for sort and filter has been added in taskRequest (**Under feature flag**)

### Before
Screenshot from 2023-10-17 20-55-48

### After
Screenshot from 2023-10-17 20-55-03

[Screencast from 19-10-23 10:20:28 PM IST.webm](https://github.com/Real-Dev-Squad/website-dashboard/assets/48379860/8ff569e0-c197-431d-8103-2497b8b373a1)

### Test Coverage
Screenshot from 2023-10-18 15-13-42

### CODE REVIEW

**Review by iamitprakash — CHANGES_REQUESTED** [review]
please use proper PR description format

**Comment by meanwhile7** [comment]
> @meanwhile7 please add tests stats or coverage

added coverage screenshot

File: taskRequests/script.js [hunk 1]
```diff
@@ -84,6 +92,178 @@ function openTaskDetails(id) {
...
+    createTimeDesc,
+  ];
+  let selectedButton = null;
+  function toggleSortModal() {
+    sortModal.classList.toggle('hidden');
+  }
+
+  function selectButton(button) {
+    if (selectedButton === button) {
+      console.log('same');
```
Comment (Achintya-Chatterjee) [inline taskRequests/script.js:172]: Why you used console here

File: taskRequests/script.js [hunk 2]
```diff
@@ -84,6 +92,178 @@ function openTaskDetails(id) {
...
+      // Clicked on the same button again, deselect it
+      selectedButton.classList.remove('selected');
+      selectedButton = null;
+      toggleSortModal();
+    } else {
+      console.log('diff');
```
Comment (Achintya-Chatterjee) [inline taskRequests/script.js:178]: Same , why you used console here

File: taskRequests/script.js (hunk 1 above)
Comment (meanwhile7) [reply taskRequests/script.js]: removed console log

File: taskRequests/script.js (hunk 2 above)
Comment (meanwhile7) [reply taskRequests/script.js]: removed console log

File: taskRequests/constants.js [hunk 3]
```diff
@@ -2,6 +2,20 @@ const taskRequestStatus = {
...
+const APPLY_FILTER_BUTTON = 'apply-filter-button';
+const SEARCH_ELEMENT = 'assignee-search';
+const SORT_BUTTON = '.sort-button';
+const OLD_FILTER = '.container__filters';
+const FILTER_CONTAINER = '.sort-filters';
+const SORT_MODAL = 'sort-modal';
+const ASSIGNEE_COUNT = 'ASSIGNEE_COUNT';
+const ASSIGNEE_Desc = 'ASSIGNEE_COUNT-desc';
+const CREATED_TIME = 'CREATED_TIME';
+const CREATED_TIME_Desc = 'CREATED_TIME-desc';
```
Comment (pallabez) [inline taskRequests/constants.js:18]:
```suggestion
const ASSIGNEE_DESC = 'ASSIGNEE_COUNT-desc';
const CREATED_TIME = 'CREATED_TIME';
const CREATED_TIME_DESC = 'CREATED_TIME-desc';
```

File: taskRequests/script.js [hunk 4]
```diff
@@ -84,6 +92,174 @@ function openTaskDetails(id) {
...
+sortButton.addEventListener('click', async (event) => {
+  event.stopPropagation();
+  sortModal.classList.toggle('hidden');
+  sortModalButtons();
+});
+
+window.onclick = function () {
+  sortModal.classList.add('hidden');
+  filterModal.classList.add('hidden');
+};
```
Comment (pallabez) [inline taskRequests/script.js:111]: Can we use `window.addEventListener` here?

File: taskRequests/script.js [hunk 5]
```diff
@@ -84,6 +92,174 @@ function openTaskDetails(id) {
...
+  ];
+
+  for (let i = 0; i < statusList.length; i++) {
+    const { name, id } = statusList[i];
+    addCheckbox(name, id, 'status-filter');
+  }
+  for (let i = 0; i < requestList.length; i++) {
+    const { name, id } = requestList[i];
+    addCheckbox(name, id, 'status-request');
+  }
```
Comment (pallabez) [inline taskRequests/script.js:239]: Can we use `statusList.map((id, item) => {...})` here?

File: taskRequests/constants.js (hunk 3 above)
Comment (meanwhile7) [reply taskRequests/constants.js]: made the suggested changes you requested

File: taskRequests/script.js (hunk 4 above)
Comment (meanwhile7) [reply taskRequests/script.js]: implemented backdrop instead of window

File: taskRequests/script.js (hunk 5 above)
Comment (meanwhile7) [reply taskRequests/script.js]: yea i used map in my recent commit

File: taskRequests/script.js [hunk 6]
```diff
@@ -84,6 +93,173 @@ function openTaskDetails(id) {
...
+      iconPathAsc: '/taskRequests/assets/sort-up.svg',
+      iconPathDesc: '/taskRequests/assets/sort-down.svg',
+    },
+    {
+      name: 'Created Time',
+      id: 'CREATED_TIME',
+      iconPathAsc: '/taskRequests/assets/sort-up.svg',
+      iconPathDesc: '/taskRequests/assets/sort-down.svg',
+    },
+  ];
```
Comment (Ajeyakrishna-k) [inline taskRequests/script.js:254]: Move these to constants file.

File: taskRequests/script.js [hunk 7]
```diff
@@ -84,6 +93,173 @@ function openTaskDetails(id) {
...
+});
+
+sortModal.addEventListener('click', (event) => {
+  event.stopPropagation();
+});
+
+filterButton.addEventListener('click', (event) => {
+  filterModal.classList.toggle('hidden');
+  backDrop.style.display = 'flex';
+  event.stopPropagation();
```
Comment (Ajeyakrishna-k) [inline taskRequests/script.js:127]: Is this required? as you already have the same in line [117](https://github.com/Real-Dev-Squad/website-dashboard/pull/576/files#diff-23b61ff1789eab0a1640feae99a0bf3465052f6c0cdda786aa4ae08c876715b5R117)

File: taskRequests/script.js [hunk 8]
```diff
@@ -84,6 +93,173 @@ function openTaskDetails(id) {
...
 
+if (isDev) {
+  containerFilter.style.display = 'flex';
+  oldContainerFilter.style.display = 'none';
+} else {
+  containerFilter.style.display = 'none';
+  oldContainerFilter.style.display = 'flex';
+}
+sortButton.addEventListener('click', async (event) => {
+  event.stopPropagation();
```
Comment (Ajeyakrishna-k) [inline taskRequests/script.js:104]: Re: is this required? line [121] (https://github.com/Real-Dev-Squad/website-dashboard/pull/576/files#diff-23b61ff1789eab0a1640feae99a0bf3465052f6c0cdda786aa4ae08c876715b5R121)

**Comment by Ajeyakrishna-k** [comment]
We also need to pass the inputs from these modals to fetch task requests and render the list again. Will it be done with this issue or a different one?

**Comment by meanwhile7** [comment]
> We also need to pass the inputs from these modals to fetch task requests and render the list again. Will it be done with this issue or a different one?

integration pr not this pr after backend is done integration pr will be created

File: taskRequests/script.js (hunk 7 above)
Comment (meanwhile7) [reply taskRequests/script.js]: fixed

File: taskRequests/script.js (hunk 8 above)
Comment (meanwhile7) [reply taskRequests/script.js:104]: thanks for reminding got it removed

File: taskRequests/script.js (hunk 6 above)
Comment (meanwhile7) [reply taskRequests/script.js:245]: I am trying to move but issue are arising

**Review by pallabez — APPROVED** [review]

**Review by kotesh-arya — APPROVED** [review]

**Review by Ajeyakrishna-k — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**